tests/test_noqa.py

```diff
@@ -0,0 +1,274 @@
+import pytest
+
+from vulture.core import NOQA_REGEXP, _parse_error_codes
+from . import check, v
+
+assert v  # Silence pyflakes.
+
+
+@pytest.mark.parametrize(
+    "line, codes",
+    [
+        ("# noqa", ["all"]),
+        ("## noqa", ["all"]),
+        ("# noqa Hi, go on.", ["all"]),
+        ("# noqa: V101", ["V101"]),
+        ("# noqa: V101, V106", ["V101", "V106"]),
+        ("# NoQA: V101,      V103, \t V104", ["V101", "V103", "V104"]),
+    ],
+)
+def test_noqa_regex_present(line, codes):
+    match = NOQA_REGEXP.search(line)
+    parsed = _parse_error_codes(match)
+    assert parsed == codes
+
+
+@pytest.mark.parametrize(
+    "line",
+    [
+        ("# noqa: 123V"),
+        ("# noqa explaination: V012"),
+        ("# noqa: ,V101"),
+        ("# noqa: #noqa: V102"),
+        ("# noqa: # noqa: V102"),
+    ],
+)
+def test_noqa_regex_no_groups(line):
+    assert NOQA_REGEXP.search(line).groupdict()["codes"] is None
+
+
+@pytest.mark.parametrize(
+    "line",
+    [("#noqa"), ("##noqa"), ("# n o q a"), ("#NOQA"), ("# Hello, noqa")],
+)
+def test_noqa_regex_not_present(line):
+    assert bool(NOQA_REGEXP.search(line)) is False
+
+
+def test_noqa_without_codes(v):
+    v.scan(
+        """\
+import this  # noqa
+
+class Cellar:  # noqa
+    @property
+    def wine(self, grapes): # noqa
+        pass
+
+    def serve(self, quantity=50):  # noqa
+        self.quantity_served = quantity  # noqa
+        return
+        self.pour()  # noqa
+"""
+    )
+    check(v.unused_attrs, [])
+    check(v.unused_classes, [])
+    check(v.unused_funcs, [])
+    check(v.unused_imports, [])
+    check(v.unused_props, [])
+    check(v.unreachable_code, [])
+    check(v.unused_vars, [])
```

Comment:
Can you use exactly one noqa for each category?

Reply:
Do you mean, break it down into individual tests for all of them?

Reply:
No, I was just surprised to see 6 noqas but 7 checks. Adding the `year` variable here as well should resolve this.

Reply:
Aah, okay. I took care of this.